packages/funding-rate-proposer/src/proposer.js

```diff
@@ -375,11 +357,35 @@ class FundingRateProposer {
   async _getContractState(contractAddress) {
     let perpetualContract = this.contractCache[contractAddress].contract;
 
-    // Grab on-chain state in parallel.
-    const [currentFundingRateData, configStoreAddress] = await Promise.all([
-      perpetualContract.methods.fundingRate().call(),
-      perpetualContract.methods.configStore().call()
-    ]);
+    let fundingRateData, configStoreAddress;
+    if (!this.multicallContractAddress) {
+      // If no multicontract address set, then read the contract's on-chain funding rate data. Note that this will not
+      // take into account any pending funding rates that will be published on the next contract interaction.
+      [fundingRateData, configStoreAddress] = await Promise.all([
+        perpetualContract.methods.fundingRate().call(),
+        perpetualContract.methods.configStore().call()
+      ]);
+    } else {
```

Comment:
Why can't we just assume that we always have a multicall address or explicitly require it?

Reply:
Oh I like asserting that `multicallContractAddress` is passed via construction I'll add